### PR TITLE
DataHub on EKS

### PR DESCRIPTION
**Pull request by bbgu1**
### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.

Adds blueprint to provision DataHub on EKS, using OpenSearch, RDS MySQL, and MSK as storage

### Motivation

[#101 ](https://github.com/awslabs/data-on-eks/issues/101)

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [x] Mandatory for new blueprints. Yes, I have added a example to support my blueprint PR
- [x] Mandatory for new blueprints. Yes, I have updated the `website/docs` or `website/blog` section for this feature
- [ ] Yes, I ran `pre-commit run -a` with this PR. Link for installing [pre-commit](https://pre-commit.com/) locally

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

**Review by askulkarni2 — CHANGES_REQUESTED** [review]
Please use blueprints v5v structure.

File: analytics/terraform/datahub-on-eks/README.md
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/README.md:2]: change this link to https://awslabs.github.io/data-on-eks/docs/data-analytics/datahub-on-eks

File: analytics/terraform/datahub-on-eks/addons.tf
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/addons.tf:5]: Let's add the version to this module.

File: analytics/terraform/datahub-on-eks/addons.tf
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/addons.tf:64]: nit:remove extra line

File: analytics/terraform/datahub-on-eks/addons.tf
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/addons.tf:21]: remove these commented lines

File: analytics/terraform/datahub-on-eks/datahub-eks/main.tf
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/datahub-eks/main.tf:34]: nit:remove extra lines

File: analytics/terraform/datahub-on-eks/datahub-eks/main.tf
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/datahub-eks/main.tf:91]: Any reason why the version is commented?

File: analytics/terraform/datahub-on-eks/datahub-eks/modules/prereq/main.tf
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/datahub-eks/modules/prereq/main.tf:76]: nite:remove this extra line

File: analytics/terraform/datahub-on-eks/datahub-eks/modules/prereq/main.tf
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/datahub-eks/modules/prereq/main.tf:97]: nite:remove this extra line

File: analytics/terraform/datahub-on-eks/datahub-eks/modules/prereq/main.tf
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/datahub-eks/modules/prereq/main.tf:115]: nite:remove this extra line

File: analytics/terraform/datahub-on-eks/datahub-eks/modules/prereq/main.tf
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/datahub-eks/modules/prereq/main.tf:131]: nite:remove this extra line

File: analytics/terraform/datahub-on-eks/datahub-eks/modules/prereq/main.tf
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/datahub-eks/modules/prereq/main.tf:192]: nite:remove this extra line

File: analytics/terraform/datahub-on-eks/datahub-eks/modules/prereq/outputs.tf
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/datahub-eks/modules/prereq/outputs.tf:35]: Its not a good idea to output the passwords. Could you write these passwords to AWS secrets manager and ask users to fetch these passwords from secrets manager?

File: analytics/terraform/datahub-on-eks/datahub-eks/modules/prereq/variables.tf
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/datahub-eks/modules/prereq/variables.tf:33]: nite:remove this extra line

File: analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml:7]: Users should be able to just increment the Helm version so that they can get the latest image related to the version. Use only modified values and ignore all the defaults such as `tag: "v0.8.45"` etc.

File: analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml:13]: Same as above comment. Don't hardcode the tag.

File: analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml:19]: We need a note to users wherever we use internet-facing load balancers saying that this is only for demo purpose. Users should change this when they run in prod

File: analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml:31]: same as above comment related to hardcoded tags

File: analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml:42]: same as above comment related to hardcoded tags

File: analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml:47]: same as above comment related to hardcoded tags

File: analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml:53]: same as above comment related to hardcoded tags

File: analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml:59]: same as above comment related to hardcoded tags

File: analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml:73]: same as above comment related to hardcoded tags

File: analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml:84]: same as above comment related to hardcoded tags

File: analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml:95]: same as above comment related to hardcoded tags

File: analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml:166]: Write a note on why you commented this section. Is this required?

File: analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml:217]: write a note on why you commented this or remove it

File: analytics/terraform/datahub-on-eks/datahub-eks/variables.tf
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/datahub-eks/variables.tf:45]: nit:remove extra line

File: analytics/terraform/datahub-on-eks/datahub.tf
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/datahub.tf:3]: rename this folder from `datahub-eks` to `datahub`. We already have top level folder saying datahub-on-eks

File: analytics/terraform/datahub-on-eks/helm-values/kubecost-values.yaml
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/helm-values/kubecost-values.yaml:24]: Does this work with Kube Prometheus Stack addon? Have you verified KubeCost from WebUI? You can remove this add-on if its not important for Datahub blueprint

File: analytics/terraform/datahub-on-eks/helm-values/prometheus-values.yaml
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/helm-values/prometheus-values.yaml:1]: Are you using this values with kube prometheus stack?

File: analytics/terraform/datahub-on-eks/install.sh
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/install.sh:16]: Do you need to this resource as well?   "aws_eks_addon.vpc_cni"

File: analytics/terraform/datahub-on-eks/locals.tf
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/locals.tf:13]: nit:remove extra lines

File: analytics/terraform/datahub-on-eks/outputs.tf
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/outputs.tf:56]: remove this if its not required

File: analytics/terraform/datahub-on-eks/variables.tf
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/variables.tf:50]: remove extra line

File: analytics/terraform/datahub-on-eks/variables.tf
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/variables.tf:61]: remove these variables and deploy these by default

File: analytics/terraform/datahub-on-eks/versions.tf
Comment (vara-bonthu) [inline analytics/terraform/datahub-on-eks/versions.tf:35]: change the key value to "e2e/datahub-on-eks/terraform.tfstate"

**Review by vara-bonthu — COMMENTED** [review]
@bbgu1 Thanks for this PR 👍🏼  PR looks good to me. You can convert this to actual PR from draft.  I left few comments which needs addressing. We will also wait for another review before merging the PR.

File: analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml
Comment (bbgu1) [reply analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml:7]: it was copied from an older version of the datahub helm example.  removed

File: analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml
Comment (bbgu1) [reply analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml:166]: example / attributes for items in extraSidecars[]

File: analytics/terraform/datahub-on-eks/helm-values/kubecost-values.yaml
Comment (bbgu1) [reply analytics/terraform/datahub-on-eks/helm-values/kubecost-values.yaml:24]: kubecost addon had been removed before.  removed the helm value file now

File: analytics/terraform/datahub-on-eks/datahub-eks/modules/prereq/outputs.tf
Comment (bbgu1) [reply analytics/terraform/datahub-on-eks/datahub-eks/modules/prereq/outputs.tf]: the module outputs are not shown to user - they are used in the parent DataHub module to create kubernetes secrets.   I will change to use external secrets instead.

File: analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml
Comment (bbgu1) [reply analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml]: removed tags from individual services but still set the 'global' version to 0.8.45, as there seemed to be issue with the latest 0.10.x version working with OpenSearch.   Will remove the version line from the value file once the issue is resolved.

File: analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml
Comment (bbgu1) [reply analytics/terraform/datahub-on-eks/datahub-eks/values/datahub_values.yaml]: done - added comment in the doc along with the value section that needs to be changed

File: analytics/terraform/datahub-on-eks/datahub.tf
Comment (bbgu1) [reply analytics/terraform/datahub-on-eks/datahub.tf]: still getting error when the folder/module name is the same as the Chart name.  Changed the folder name to datahub-addon.

File: analytics/terraform/datahub-on-eks/install.sh
Comment (bbgu1) [reply analytics/terraform/datahub-on-eks/install.sh:16]: added dependency to the addon module to this resource to have the cni deployed before other addons.

**Review by vara-bonthu — COMMENTED** [review]
@bbgu1 This PR looks good to me. One change required for docs before merging. Could you please move the docs under "website/docs/blueprints/data-analytics".

**Comment by bbgu1** [comment]
> @bbgu1 This PR looks good to me. One change required for docs before merging. Could you please move the docs under "website/docs/blueprints/data-analytics".

Done

**Review by vara-bonthu — APPROVED** [review]
LGTM 🚀

**Pull request merged**
